sql-plugin/src/main/scala/org/apache/spark/sql/rapids/HashFunctions.scala

```diff
@@ -108,6 +108,31 @@ case class GpuXxHash64(children: Seq[Expression], seed: Long) extends GpuHashExp
   }
 }
 
+object XxHash64Utils {
+  /**
+   * Compute the max stack size that `inputType` will use,
+   * refer to the function `check_nested_depth` in src/main/cpp/src/xxhash64.cu
+   * in spark-rapids-jni repo.
+   * Note:
+   * - This should be sync with `check_nested_depth`
+   * - Map in cuDF is list of struct
+   *
+   * @param inputType the input type
+   * @return the max stack size that inputType will use for this input type.
+   */
+  def computeMaxStackSize(inputType: DataType): Int = {
+    inputType match {
+      case ArrayType(c: StructType, _) => 1 + computeMaxStackSize(c)
+      case ArrayType(c: DataType, _) => computeMaxStackSize(c)
```

Comment:
Can this be combined like this?
```suggestion
      case ArrayType(c: _, _) => computeMaxStackSize(c)
```

Reply:
No,  in JNI, list(other type) is optmized to use less stack depth, only use depthOf(other).
But for list(struct), should be 1 + depthOf(struct).

Reply:
NIT:
```suggestion
      case ArrayType(c: StructType, _) => 1 + computeMaxStackSize(c)
      case ArrayType(c, _) => computeMaxStackSize(c)
```